Feature/lecture list

park3hho opened the pull request:
## ✅ PR 요약
- 작업 요약:  lectures app 생성 및 models 작성, user_model과 연동. 

## 📄 상세 내용
- [1] lectures app 생성 
```
├── lectures/
│   ├── migrations/     # 마이그레이션 파일
│   ├── services/       # 앱에서 사용되는 서비스 로직을 구현하는 폴더 / 서비스 로직
│   ├── repository/     # 앱의 데이터베이스 모델에 관련된 CRUD 동작을 정의하는 폴더 / ORM
│   ├── tests/          # 앱에서 사용되는 테스트들을 구현하는 폴더
│   ├── models/         # 앱에서 사용되는 모델들을 정의하는 폴더
│   │   ├── categories/                          # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   ├── crawled_lecture_reviews/    # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   ├── lecture_bookmarks/            # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   ├── lecture_search_logs/            # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   ├── crawled_lectures/            # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   ├── user_prefer_categories/            # 앱 전용 URL 라우팅을 정의하는 폴더
│   │   └── lecture_categories/             # 앱 전용 URL 라우팅을 정의하는 폴더
│   ├── urls/              # 앱 전용 URL 라우팅을 정의하는 폴더
│   ├── serializers/    # 시리얼 라이저 모음 폴더
│   ├── views/           # CBV, FBV 를 구현하는 폴더 / HTTP 호출 관련
│   ├── admin.py      # Django Admin 관련 설정 ( 익스턴십에서는 사용하지 않음 삭제 요망 )
│   └── apps.py        # 앱 설정
```
- [2] models 작성
> 복합키 설정 완료
> BaseModel, UuidModel 사용

- [3] user_model과 연동
> 이상 없음 확인

- [4] 기타 환경 조정
> APP에 model 추가

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [1] migrate 이상 없습니다. 문제 업습니다.
- [2] mypy까지 했습니다.

park3hho committed:
# 🚚 build: 0.0.1 <models 작성 완료 및 기타 환경설정>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 🚚 models 추가
[2] 🚚 reload 추가
[3] 🚚 miration 통과 (mirate만 실행하면 됨)
[4] 🚚 기타 쓸모없는 파일 리팩토링 (삭제 추가)

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

# Conflicts:
#	config/settings/base.py

park3hho committed:
# 🚚 build: 0.0.2 <models 네이밍 통일>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 🚚 models 네이밍 통일
[2] 🚚
[3] 🚚
[4] 🚚

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

park3hho committed:
[1] ♻️ models 네이밍 통일
[2] 🚚 viwes.py 주석처리
[3] ✨ model 선언
[4] 🚚

========================================================================

park3hho committed:
# ✨♻️ build: 0.0.4 <models 네이밍 통일 / 린트했어요>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] ♻️ models 네이밍 통일
[2] 🚚 viwes.py 주석처리
[3] ✨ model 선언
[4] 🚚

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

SimJunSik commented on apps/lectures/models/crawled_lectures.py:
```diff
@@ -0,0 +1,44 @@
+    original_price = models.BigIntegerField(default=0)
+    discount_price = models.BigIntegerField(default=0)
+    url_link = models.CharField(max_length=255)
+    thumbnail_img_url = models.CharField(max_length=255, null=True, blank=True)
```
url 을 저장하는 필드니까 유효성 검증 등이 필요하니 `URLField` 를 사용하는게 좋을 것 같네요

SimJunSik commented on apps/lectures/views/lectures_list.py:
```diff
@@ -0,0 +1,12 @@
+#     # 이 쿼리셋을 이용하여 목록을 조회합니다.
+#     queryset = Lecture.objects.all()
+#     # 이 시리얼라이저를 사용하여 데이터 형식을 변환합니다.
+#     serializer_class = LectureSerializer
```
지금 바로 사용할 코드가 아니라면, 이후에 이 코드를 정말 사용하게 될 때에 PR 을 따로 만들어서 하는걸 권장드립니다

park3hho replied:
넵 확인했습니다. 
전체 피알 안날리겠습니다.

Meoyoug commented on apps/lectures/admin.py:
장고 어드민 대신 직접 어드민 페이지 작성할 예정이니 이파일은 제거해주세요!

park3hho replied:
드민이 삭제하겠숩니다 감사합니다.

Meoyoug commented on apps/lectures/models/categories.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
노어드민이라서 제가제거 했습니다.

Meoyoug commented on apps/lectures/models/crawled_lecture_reviews.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
노어드민이라서 제가제거 했습니다.

SimJunSik commented on config/wsgi.py:
```diff
@@ -1,12 +1,3 @@
-"""
-WSGI config for config project.
-
-It exposes the WSGI callable as a module-level variable named ``application``.
-
-For more information on this file, see
-https://docs.djangoproject.com/en/5.2/howto/deployment/wsgi/
-"""
-
```
PR 하나에는 가능하면 하나의 주제에 대해서만 다루는게 실무에서 많이 권장됩니다
특히나 아래 `docker-compose.local.yml` 변경사항은 lecture list 와는 관련이 없지만 동작이 바뀌는 변경사항이기 때문에 다른 PR 로 처리하는 걸 권장드려요

park3hho replied:
하나의 피알에 하나의 주제.

확인했습니다. 감사합니다.

SimJunSik replied:
PR 하나에 하나의 주제를 다루면 아래와 같은 좋은 점들이 있어서 권장되는 사항이니 참고 부탁드려요!

- 리뷰어도 하나의 주제에 대해 집중해서 리뷰 할 수 있다
- 해당 PR 의 논의 내용도 그 주제에 대해서만 이뤄지니 나중에 다른 사람이 해당 PR 을 보더라도 맥락 파악에 수월해진다
- 하나에 대해서만 다루다보면 혹시나 해당 PR 에 문제가 있을 때 딱 그 부분만 revert 할 수 있다 
등등..

Meoyoug commented on apps/lectures/models/crawled_lectures.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
제가제거했습니다

Meoyoug commented on apps/lectures/models/lecture_bookmarks.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

Meoyoug commented on apps/lectures/models/lecture_categories.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
제가제거했습니다

Meoyoug commented on apps/lectures/models/lecture_search_logs.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
제가제거했습니다

Meoyoug commented on apps/lectures/models/user_prefer_categories.py:
verbose_name, verbose_name_plural 옵션의 경우 admin, form을 사용하지않기 때문에 제거요망

park3hho replied:
제가제거했습니다

park3hho committed:
# 💡 build: 0.0.5 <PR Review 내용 반영>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 💡 URLField 적용
[2] 💡 파일 전체 PR 금지 ㅋㅋ
[3] 💡 해당 X PR 분리하기 (앞으로)
[4] 💡 어드민 파일 삭제 및 관련 코드 삭제

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

park3hho committed:
# 💡 build: 0.0.6 <앗차차마이그레이션을까먹었다.>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 💡 마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자마이그레이션을까먹지말자
[2] 💡
[3] 💡
[4] 💡

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

park3hho commented:
마이그레이션을하고올리도록하겠습니다.

park3hho committed:
# 💡 build: 0.0.7 <코드 하나 더 지워버렸다... 추가했습니다. ㅋㅋ; ㅋㅋ;>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 💡 ㅎㅎㅋㅋ,,
[2] 💡
[3] 💡
[4] 💡

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

park3hho committed:
# 💡 build: 0.0.8 <코드 하나 더 지워버렸다... 추가했습니다. ㅋㅋ; ㅋㅋ;>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 💡 ㅎㅎㅋㅋ,,
[2] 💡
[3] 💡
[4] 💡

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

park3hho commented:
블랙 안되서 해서 올렸습니다.

park3hho committed:
# 💡 build: 0.0.9 <migration을 까먹지말자. migration을 까먹지말자. migration을 까먹지말자.migration을 까먹지말자.>

# 2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
[1] 💡 migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.migration을 까먹지말자.
[2] 💡
[3] 💡
[4] 💡

# 3. 이슈 ( 선택 )
# 관련 이슈: #123
========================================================================
# Commit Convention
# ✨ feat     : 새로운 기능 추가
# 🐛 fix      : 버그 수정
# 💡 chore    : 기능 추가 없이 코드 수정 (오타, 주석 등)
# 📝 docs     : 문서 수정 (README 등)
# 🚚 build    : 빌드 관련 파일 수정
# ✅ test     : 테스트 코드 추가/변경 (프로덕션 코드 변경 없음)
# ♻️ refactor : 리팩터링 (기능 변화 없음)
# 🚑 hotfix   : 긴급 수정

Meoyoug reviewed (approved):
고생하셨습니다! @SimJunSik 님 커멘트 해결하고 머지하겠습니다!
